velox/exec/WindowPartition.cpp

```diff
@@ -364,7 +364,15 @@ void WindowPartition::updateKRangeFrameBounds(
   vector_size_t start = 0;
   vector_size_t end;
   RowColumn frameRowColumn = columns_[frameColumn];
-  RowColumn orderByRowColumn = columns_[inputMapping_[orderByColumn]];
+  column_index_t mappedOrderByColumn = -1;
+  for (auto i = 0; i < inputMapping_.size(); i++) {
+    if (inputMapping_[i] == orderByColumn) {
+      mappedOrderByColumn = i;
+      break;
+    }
+  }
+  VELOX_DCHECK_NE(mappedOrderByColumn, -1);
+  RowColumn orderByRowColumn = columns_[mappedOrderByColumn];
```

Comment:
So the problem with the original code is that orderByColumn is the column index into data_ where columns are already reordered to start with partition-by keys and sorting keys. On the other hand, columns_ keeps the original column order (i.e., same as the original input to the Window operator). So there was a mismatch. To make it worse, the original code accesses columns_ through `columns_[inputMapping_[orderByColumn]]` where inputMapping_ is a mapping from the original input column index to the column index after reordering (i.e., it is the reverse of what the code intended to do). 

Since columns_ was derived from data_ (see WindowPartition constructor), and both data_ and orderByColumn follow the column index after column reordering, I wonder whether the fix can be simplified to the following, so that you don't need to loop over inputMapping_ to find the original column index of orderByColumn.
```
RowColumn orderByRowColumn = data_->columnAt(orderByColumn);
```